kf_hungarian_tracker/kf_hungarian_tracker/kf_hungarian_node.py

```diff
@@ -98,7 +98,15 @@ def callback(self, msg):
             try:
                 trans = self.tf_buffer.lookup_transform(self.global_frame, msg.header.frame_id, rclpy.time.Time())
                 msg.header.frame_id = self.global_frame
+                # do_transform_vector3(vector, trans) resets trans.transform.translation
+                # values to 0.0, so we need to preserve them for future usage in the loop below
+                translation_backup_x = trans.transform.translation.x
```

Comment:
Instead, you can make a deep copy of the `trans` to send to that function instead which is more clean than separately setting the 3 variables independently

Reply:
Its also 6 vs 3 items, this isn't exactly a massive operation. Did you notice a tangible difference? The fact this is python too is a huge performance sink so I'm not too worried about peanuts of performance on this. The actual algorithm is so heavy this is neglibable but nice for readability (unless you noticed actual differences, in that case, sure lets do it)

Reply:
I've measured the slowdown if change "3-copy" approach to the `copy.deepcopy(trans.transform.translation)` approach.
The results are showing ~8% slowdown of TF cycle (+) time and ~2.8% slowdown of whole callback time when switching to the second approach:
| Approach  | TF cycle time, s | Whole callback time, s |
| ------------- | ------------- | ------- |
| 3-copy  | 0.11608 | 1.31565 |
| deepcopy | 0.12519 | 1.35290 |
| Slowdown, % | 7.84% | 2.83% |

So, if you don't mind, even in Python I'd like to remain current approach, because it is notable faster in this cycle.

Reply:
But over how many detections? I think we covered that the number of laser points are not an appropriate proxy for this as an unrealistic number of locally tracked discrete obstacles? 

Also, you mention in the copy that `do_transform_vector3` resets the transform, but that's called twice, so does the second just not work?

Reply:
> But over how many detections? I think we covered that the number of laser points are not an appropriate proxy for this as an unrealistic number of locally tracked discrete obstacles?

There are 360 obstacles from laserscan simply detected by laser obstacles detector, written for local kf_hungarian_tracker understanding and verification. In most scenarios, yes this is unrealistic number for objects got by more-less real detector. However, why don't use the optimal construction, when we could use it? Is the reason is 3 assignments could decrease the readability/neatness of the code? 

> Also, you mention in the copy that `do_transform_vector3` resets the transform, but that's called twice, so does the second just not work?

Not quite so: `do_transform_vector3()` does not use transform's translation at all. That is why at the beginning of the `do_transform_vector3()` all these translations are being zero-ed:
```
# Vector3Stamped
def do_transform_vector3(
        vector3: Vector3Stamped,
        transform: TransformStamped) -> Vector3Stamped:
    transform.transform.translation.x = 0.0
    transform.transform.translation.y = 0.0
    transform.transform.translation.z = 0.0
    _, point = _decompose_affine(
        np.matmul(
            _transform_to_affine(transform),
            _build_affine(translation=[
                vector3.vector.x,
                vector3.vector.y,
                vector3.vector.z
            ])))
  ...
```

This was made intentionally in `tf2_geometry_msgs.py` by:
```
commit b0b1046422ba757ee8a5041474d25bf423e839df
Author: chapulina <burajiru.no.chapulina@gmail.com>
Date:   Wed Dec 9 11:46:34 2015 -0800

    vector3 not affected by translation
```
And it is not clear why it is still in the RCLPY API. I think, it might be for performance reasons.

Returning back to the discussion: `do_transform_vector3()` do not use input transform and resets it, but `do_transform_point(p, trans)` (which is also in the transformation loop) uses it. So, before the `do_transform_point()` we need to restore `trans.transform.translation` to do everything correctly. Otherwise, only the first point from `detections` list will be handled correctly.

Reply:
> Is the reason is that 3 assignments could decrease the readability/neatness of the code?

At this point, I don't care about that - I'm more asking the second question on calling the function 2x when 1st will reset some important stuff.